spyder/utils/encoding.py

```diff
@@ -249,4 +249,7 @@ def is_text_file(filename):
     """
     Test if the given path is a text-like file.
     """
-    return not is_binary(filename)
+    try:
+        return not is_binary(filename)
+    except IsADirectoryError:

```

Comment:
In Python2 `IOError` will be raised https://docs.python.org/2.7/library/exceptions.html#exceptions.IOError

I don't know how is the best way to deal with different exceptions and python2 and python3

should I use `PY2`?:
```
if PY2:
    try:
        return not is_binary(filename)
    except IOError:
        return False
else
    try:
        return not is_binary(filename)
    except IsADirectoryError:
        return False
```

or just catch any exception:
```
    try:
        return not is_binary(filename)
    except:
        return False
```